common-custom-user-data-maven-extension/src/main/java/com/gradle/CommonCustomUserDataGradleEnterpriseListener.java

```diff
@@ -1,60 +1,54 @@
 package com.gradle;
 
 import com.gradle.maven.extension.api.GradleEnterpriseApi;
+import com.gradle.maven.extension.api.GradleEnterpriseListener;
 import com.gradle.maven.extension.api.cache.BuildCacheApi;
 import com.gradle.maven.extension.api.scan.BuildScanApi;
-import org.apache.maven.AbstractMavenLifecycleParticipant;
 import org.apache.maven.MavenExecutionException;
 import org.apache.maven.execution.MavenSession;
-import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
 import org.codehaus.plexus.logging.Logger;
 
-import javax.inject.Inject;
+import javax.annotation.Nullable;
 
+@SuppressWarnings("unused")
 @Component(
-    role = AbstractMavenLifecycleParticipant.class,
-    hint = "common-custom-user-data",
-    description = "Captures common custom user data in Maven build scans"
+        role = GradleEnterpriseListener.class,
+        hint = "common-custom-user-data",
+        description = "Captures common custom user data in Maven build scans"
 )
-public final class CommonCustomUserDataMavenExtension extends AbstractMavenLifecycleParticipant {
+public final class CommonCustomUserDataGradleEnterpriseListener implements GradleEnterpriseListener {
 
-    private final PlexusContainer container;
-    private final Logger logger;
-
-    @Inject
-    public CommonCustomUserDataMavenExtension(PlexusContainer container, Logger logger) {
-        this.container = container;
-        this.logger = logger;
-    }
+    @Requirement
+    private Logger logger;
 
     @Override
-    public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
-        logger.debug("Executing extension: " + getClass().getSimpleName());
-
-        GradleEnterpriseApi gradleEnterprise = ApiAccessor.lookupGradleEnterpriseApi(container, getClass());
-        if (gradleEnterprise != null) {
+    public void configure(GradleEnterpriseApi api, @Nullable MavenSession session) {
```

Comment:
Is there not a way to implement the caller of `configure` such that `MavenSession` is never null? 
Forcing the client to deal with this seems unfortunate.

Reply:
I just made changes on https://github.com/gradle/dotcom/pull/10070 and here, as a consequence, to do just that. It's indeed better.